diff -ruN xv6-public/Drawtest.c xv6/Drawtest.c
--- xv6-public/Drawtest.c	1970-01-01 05:30:00.000000000 +0530
+++ xv6/Drawtest.c	2022-08-12 22:30:17.862524972 +0530
@@ -0,0 +1,11 @@
+#include "types.h"
+#include "stat.h"
+#include "user.h"
+
+int main(void)
+{
+    static char buf[2000];
+    printf(1, "Draw syscall returns %d\n", draw((void*)buf, 2000));
+    printf(1, "%s", buf);
+    exit();
+}
\ No newline at end of file
diff -ruN xv6-public/Makefile xv6/Makefile
--- xv6-public/Makefile	2022-08-12 22:49:57.639480096 +0530
+++ xv6/Makefile	2022-08-12 15:17:26.907658813 +0530
@@ -181,6 +181,7 @@
 	_usertests\
 	_wc\
 	_zombie\
+	_Drawtest\
 
 fs.img: mkfs README $(UPROGS)
 	./mkfs fs.img README $(UPROGS)
@@ -253,6 +254,7 @@
 	printf.c umalloc.c\
 	README dot-bochsrc *.pl toc.* runoff runoff1 runoff.list\
 	.gdbinit.tmpl gdbutil\
+	Drawtest.c\
 
 dist:
 	rm -rf dist
diff -ruN xv6-public/syscall.c xv6/syscall.c
--- xv6-public/syscall.c	2022-08-12 22:49:57.643480083 +0530
+++ xv6/syscall.c	2022-08-12 22:28:28.151872053 +0530
@@ -103,6 +103,7 @@
 extern int sys_wait(void);
 extern int sys_write(void);
 extern int sys_uptime(void);
+extern int sys_draw(void);
 
 static int (*syscalls[])(void) = {
 [SYS_fork]    sys_fork,
@@ -126,6 +127,7 @@
 [SYS_link]    sys_link,
 [SYS_mkdir]   sys_mkdir,
 [SYS_close]   sys_close,
+[SYS_draw]    sys_draw,
 };
 
 void
diff -ruN xv6-public/syscall.h xv6/syscall.h
--- xv6-public/syscall.h	2022-08-12 22:49:57.643480083 +0530
+++ xv6/syscall.h	2022-08-12 22:28:00.679871949 +0530
@@ -20,3 +20,4 @@
 #define SYS_link   19
 #define SYS_mkdir  20
 #define SYS_close  21
+#define SYS_draw   22
\ No newline at end of file
diff -ruN xv6-public/sysproc.c xv6/sysproc.c
--- xv6-public/sysproc.c	2022-08-12 22:49:57.647480073 +0530
+++ xv6/sysproc.c	2022-08-12 22:28:37.439962811 +0530
@@ -89,3 +89,42 @@
   release(&tickslock);
   return xticks;
 }
+
+int
+sys_draw(void)
+{
+  void* buf;
+  uint size;
+
+  argptr(0, (void*)&buf, sizeof(buf));
+  argptr(1, (void*)&size, sizeof(size));
+
+  char text[] = " \n \
+        ****\n\
+     ********\n\
+    **  ******\n\
+     *   ******     ******\n\
+         ******   *********\n\
+          ****  *****   ***\n\
+          ***  ***     **\n\
+    *************       *\n\
+  ******************\n\
+ *****   H*****H*******\n\
+ ***     H-___-H  *********\n\
+  ***    H     H      *******\n\
+   **    H-___-H        *****\n\
+     *   H     H         ****\n\
+         H     H         ***\n\
+         H-___-H         **\n\
+         H     H         *\n\
+         H-___-H\n\
+\n\
+         ALOHA!!\n\n";
+
+  if(sizeof(text) > size)
+    return -1;
+  
+  strncpy((char*)buf, text, size);
+  return sizeof(text);
+
+}
diff -ruN xv6-public/user.h xv6/user.h
--- xv6-public/user.h	2022-08-12 22:49:57.647480073 +0530
+++ xv6/user.h	2022-08-12 22:28:59.384126744 +0530
@@ -23,6 +23,7 @@
 char* sbrk(int);
 int sleep(int);
 int uptime(void);
+int draw(void*, uint);
 
 // ulib.c
 int stat(const char*, struct stat*);
diff -ruN xv6-public/usys.S xv6/usys.S
--- xv6-public/usys.S	2022-08-12 22:49:57.647480073 +0530
+++ xv6/usys.S	2022-08-12 22:28:49.656080862 +0530
@@ -29,3 +29,4 @@
 SYSCALL(sbrk)
 SYSCALL(sleep)
 SYSCALL(uptime)
+SYSCALL(draw)
\ No newline at end of file
